operator/cmd/webhook/builder.go

```diff
@@ -0,0 +1,96 @@
+package webhook
+
+import (
+	"flag"
+
+	"github.com/keptn/lifecycle-toolkit/operator/cmd/certificates"
+	"github.com/keptn/lifecycle-toolkit/operator/cmd/config"
+	cmdManager "github.com/keptn/lifecycle-toolkit/operator/cmd/manager"
+	"github.com/keptn/lifecycle-toolkit/operator/webhooks"
+	"github.com/keptn/lifecycle-toolkit/operator/webhooks/pod_mutator"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	FlagCertificateDirectory   = "certs-dir"
+	FlagCertificateFileName    = "cert"
+	FlagCertificateKeyFileName = "cert-key"
+)
+
+var (
+	certificateDirectory   string
+	certificateFileName    string
+	certificateKeyFileName string
+)
+
+type Builder struct {
+	configProvider  config.Provider
+	managerProvider cmdManager.Provider
+	namespace       string
+	podName         string
+}
+
+func NewWebhookBuilder() Builder {
+	return Builder{}
+}
+
+func (builder Builder) SetConfigProvider(provider config.Provider) Builder {
+	builder.configProvider = provider
+	return builder
+}
+
+func (builder Builder) SetManagerProvider(provider cmdManager.Provider) Builder {
+	builder.managerProvider = provider
+	return builder
+}
+
+func (builder Builder) SetNamespace(namespace string) Builder {
+	builder.namespace = namespace
+	return builder
+}
+
+func (builder Builder) SetPodName(podName string) Builder {
+	builder.podName = podName
+	return builder
+}
+
+func (builder Builder) GetManagerProvider() cmdManager.Provider {
+	if builder.managerProvider == nil {
+		builder.managerProvider = NewWebhookManagerProvider(certificateDirectory, certificateKeyFileName, certificateFileName)
+	}
+
+	return builder.managerProvider
+}
+
+func (builder Builder) Run(webhookManager manager.Manager) error {
+
+	addFlags()
+	builder.GetManagerProvider().SetupWebhookServer(webhookManager)
+
+	certificates.
+		NewCertificateWatcher(webhookManager, builder.namespace, webhooks.SecretCertsName, ctrl.Log.WithName("Webhook Cert Manager")).
+		WaitForCertificates()
```

Comment:
since this could last for 5min, should we do something for the health/ready probes?

Reply:
I will add this in a TODO list pr together with docs